api/src/db/foreign/dialect.py

```diff
@@ -0,0 +1,42 @@
+#
+# Support for generating SQL for "CREATE FOREIGN TABLE".
+#
+
+import re
+
+import sqlalchemy
+
+
+class ForeignTableDDLCompiler(sqlalchemy.sql.compiler.DDLCompiler):
+    """SQLAlchemy compiler for creating foreign tables."""
+
+    def create_table_constraints(self, _table, _include_foreign_key_constraints=None, **kw):
```

Comment:
Missing type definitions on a few of these. FIne to just set them to `Any` if they're complex internal types to SQLAlchemy and not just str/bool or similar. A quick check of the base class, it doesn't look to type them inline, so not a lot to go on.